File: tools/bench/answers/go_haiku/s2/01_rle.go
package main

import (
	"fmt"
)

func main() {
	s := "aaabbcccca"
	var result string
	i := 0
	for i < len(s) {
		char := s[i]
		count := 1
		for i+count < len(s) && s[i+count] == char {
			count++
		}
		result += fmt.Sprintf("%c%d", char, count)
		i += count
	}
	fmt.Println(result)
}

File: tools/bench/answers/rust_haiku/s3/01_rle.rs
fn main() {
    let s = "aaabbcccca";
    let mut result = String::new();
    let mut i = 0;
    let chars: Vec<char> = s.chars().collect();

    while i < chars.len() {
        let ch = chars[i];
        let mut count = 1;
        while i + count < chars.len() && chars[i + count] == ch {
            count += 1;
        }
        result.push(ch);
        result.push_str(&count.to_string());
        i += count;
    }

    println!("{}", result);
}
